src/hooks/use-is-valid-blueprint.ts

```diff
@@ -0,0 +1,42 @@
+import { useEffect, useState, useMemo } from 'react';
+import { getIpcApi } from 'src/lib/get-ipc-api';
+import type { Blueprint } from 'src/stores/wpcom-api';
+
+export function useIsValidBlueprint( className?: string, content?: string ) {
+	const blueprintJson = useMemo< Blueprint[ 'blueprint' ] | null >( () => {
+		if ( ! className?.includes( 'language-json' ) ) {
+			return null;
+		}
+		if ( ! content ) {
+			return null;
+		}
+		try {
+			const parsed = JSON.parse( content );
+			return parsed;
+		} catch {
+			return null;
+		}
+	}, [ className, content ] );
+
+	const [ isBlueprintValid, setIsBlueprintValid ] = useState( false );
+
+	useEffect( () => {
+		const validate = async () => {
+			setIsBlueprintValid( false );
```

Comment:
I am wondering do we need to set state here again as default value is already been set above?